codec/src/main/java/io/netty/handler/codec/MessageAggregator.java

```diff
@@ -221,7 +214,24 @@ public void operationComplete(ChannelFuture future) throws Exception {
                         }
                     };
                 }
-                ctx.writeAndFlush(continueResponse).addListener(listener);
+
+                // Make sure to call this before writing, otherwise reference counts may be invalid
+                boolean closeAfterWrite = closeAfterContinueResponse(continueResponse);
+                handlingOversizedMessage = ignoreContentAfterContinueResponse(continueResponse);
```

Comment:
Move this after the the `if (closeAfterWrite) { }` block as we may not need to do this at all